encoding/thrift/inbound_nowire_test.go

```diff
@@ -0,0 +1,374 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package thrift
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/thriftrw/protocol/binary"
+	"go.uber.org/thriftrw/protocol/stream"
+	streamtest "go.uber.org/thriftrw/thrifttest/stream"
+	"go.uber.org/thriftrw/wire"
+	"go.uber.org/yarpc/api/transport/transporttest"
+	"go.uber.org/yarpc/internal/testtime"
+)
+
+const _body = "decoded"
+
+type bodyReader struct {
+	body string
+	err  error
+}
+
+func (br *bodyReader) Decode(sr stream.Reader) error {
+	br.body = _body
+	return br.err
+}
+
+type responseEnveloper struct {
+	name         string
+	envelopeType wire.EnvelopeType
+}
+
+var _ stream.Enveloper = (*responseEnveloper)(nil)
+
+func (re responseEnveloper) MethodName() string              { return re.name }
+func (re responseEnveloper) EnvelopeType() wire.EnvelopeType { return re.envelopeType }
+func (re responseEnveloper) Encode(stream.Writer) error      { return nil }
```

Comment:
Why are we implementing a stub for `stream.Enveloper` instead of using `MockEnveloper` from streamtest?

Reply:
I would prefer the Mock for the following reasons:
- Yes readability is a bit harder, but mock are known by most of the developers, no need of understand what the stub is doing.
- More control IMO, for instance you could do `env.EXPECT().MethodName().Return("foo").Times(1)`. This IMO adds more control on what the test should actually test.

Reply:
Done.